src/Mvc/test/Mvc.FunctionalTests/ApiBehaviorTest.cs

```diff
@@ -364,4 +332,132 @@ public async Task SerializingValidationProblemDetails_WithExtensionData()
                 });
         }
     }
+
+    public class ApiBehaviorTest : ApiBehaviorTestBase<BasicWebSite.StartupWithSystemTextJson>
+    {
+        public ApiBehaviorTest(MvcTestFixture<BasicWebSite.StartupWithSystemTextJson> fixture)
+            : base(fixture)
+        {
+        }
+
+        [Fact]
+        public override async Task ActionsReturnBadRequest_WhenModelStateIsInvalid()
+        {
+            // Arrange
+            using var _ = new ActivityReplacer();
+
+            var contactModel = new Contact
+            {
+                Name = "Abc",
+                City = "Redmond",
+                State = "WA",
+                Zip = "Invalid",
+            };
+            var contactString = JsonConvert.SerializeObject(contactModel);
+
+            // Act
+            var response = await Client.PostAsJsonAsync("/contact", contactModel);
+
+            // Assert
+            await response.AssertStatusCodeAsync(HttpStatusCode.BadRequest);
+            Assert.Equal("application/problem+json", response.Content.Headers.ContentType.MediaType);
+            var problemDetails = JsonConvert.DeserializeObject<ValidationProblemDetails>(
+                await response.Content.ReadAsStringAsync(),
+                new JsonSerializerSettings
+                {
+                    Converters = { new ValidationProblemDetailsConverter() }
+                });
+            Assert.Collection(
+                problemDetails.Errors.OrderBy(kvp => kvp.Key),
+                kvp =>
+                {
+                    Assert.Equal("Name", kvp.Key);
+                    var error = Assert.Single(kvp.Value);
+                    Assert.Equal("The field Name must be a string with a minimum length of 5 and a maximum length of 30.", error);
+                },
+                kvp =>
+                {
+                    Assert.Equal("Zip", kvp.Key);
+                    var error = Assert.Single(kvp.Value);
+                    Assert.Equal("The field Zip must match the regular expression '\\d{5}'.", error);
+                }
+            );
+
+            Assert.Collection(
+                problemDetails.Extensions,
+                 kvp =>
+                 {
+                     Assert.Equal("extensions", kvp.Key);
```

Comment:
This is kinda busted - https://github.com/dotnet/corefx/issues/38769. I've added a test to make sure the rest of Json is correct. If we get this in, the most we'd have to do to "react" is to fix up tests

Reply:
Should we skip this test instead? I don't really want to encode wrong behavior in tests. It's better to just skip if possible.

Put an other way, I don't really want a test that partially validates the expectations.

Reply:
Yeah, seems reasonable. It's likely https://github.com/dotnet/corefx/issues/38769 will get fixed for preview7.